### ISSUE TITLE
React ref as string throw an error

### ISSUE THREAD

**Issue by Inateno** [storybooks/storybook#3424]
Hi !

If I use the React refs with string (old way) I got an error:
![image](https://user-images.githubusercontent.com/1068709/38813111-6d2e0ea6-418e-11e8-8030-3f80aa5c1280.png)

But then if I use the !["new" way of doing it](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) it's okay.

The fact is, in my application refs as string are still okay (and running). React team removed the string refs and didn't set it as deprecated yet too.

Seems like an issue with storybook himself.

I am using react@16.3.1.

Here is a snippet code to reproduce the error, change the ref string by an instanciated ref and it will work properly.

``` javascript
import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './styles.css'

export default class ExampleComponent extends Component {
  static propTypes = {
    text: PropTypes.string
  }

  render() {
    const {
      text
    } = this.props

    return (
      <div className={styles.test} ref="toto">
        Example Component: {text}
      </div>
    )
  }
}
```

**Comment by Hypnosphi** [comment]
> didn't set it as deprecated yet.

That's only partially true
https://reactjs.org/docs/strict-mode.html#warning-about-legacy-string-ref-api-usage

**Comment by Hypnosphi** [comment]
But the actual issue is probably that:
> You have multiple copies of React loaded

What does this command output for you?

```
npm list react
```

**Comment by Hypnosphi** [comment]
Can you please share some minimal part of your configuration as a GitHub repo, co that I could debug it?

**Comment by Inateno** [comment]
Ok after creating an independent project I found the "real" problem.

So ref as string is working in storybook, but it is not 2 different versions of React who create the problem.
It is the same version of React loaded twice.

What is interesting is, the ref instance works anyway, the ref string doesn't.

Here is the link to the repo, will be more simple to understand: https://github.com/Inateno/storybook-ref-bug

I'm trying since this morning to figure out how could I prevent this, maybe in the custom webpack file by adding a rule to ignore sub node_modules, idk yet.

**Comment by Inateno** [comment]
Yeah. Because everything in package is an independent library that needs React to launch example and build.
We are using Lerna on top of all those things.

And even if there is a way to say to rollup (or else) to use dependencies from the parent folder, this will mean that all packages must have the same dependencies which can be possible but is not true all the time.

**Comment by Hypnosphi** [comment]
OK, that case is pretty similar to https://github.com/storybooks/storybook/issues/3381

The possible solutions are pretty much the same:

— Use [yarn workspaces](https://yarnpkg.com/blog/2017/08/02/introducing-workspaces/) for bootstrapping phase
— [Hoist common `devDependencies`](https://github.com/lerna/lerna#common-devdependencies) manually
— Create a separate package with storybook and tests for your library, and have react `devDependency` only there

If you're going to have different packages using different React versions, currently you have no other option than to use separate storybooks for them
